compositor_render/src/wgpu/format/pixel_yuv_to_rgba.wgsl

```diff
@@ -0,0 +1,52 @@
+struct VertexInput {
+    @location(0) position: vec3<f32>,
+    @location(1) tex_coords: vec2<f32>,
+}
+
+struct VertexOutput {
+    @builtin(position) position: vec4<f32>,
+    @location(0) tex_coords: vec2<f32>,
+}
+
+@vertex
+fn vs_main(input: VertexInput) -> VertexOutput {
+    var output: VertexOutput;
+
+    output.position = vec4(input.position, 1.0);
+    output.tex_coords = input.tex_coords;
+
+    return output;
+}
+
+@group(0) @binding(0) var texture: texture_2d<f32>;
+@group(1) @binding(0) var sampler_: sampler;
+
+@fragment
+fn fs_main(input: VertexOutput) -> @location(0) vec4<f32> {
+    var dimensions = textureDimensions(texture);
+    var eps = 0.0001;
+    var half_pixel_width = 0.5 / f32(dimensions.x);
+
+    // x_pos represents index of a column(pixel) on the output texture
+    // - dimensions.x represents half of the output width, so we need multiply * 2
+    // - input.tex_coords represents middle of pixel, so to shift to column index we need to shift by that value
+    // - adding eps to avoid numerical errors when converting f32 -> u32
+    var x_pos = u32((input.tex_coords.x * f32(dimensions.x) - half_pixel_width + eps) * 2.0);
+    // x_pos/2 is calculated before conversion to float to make sure that reminder is lost for odd column.
+    var tex_coords = vec2((f32(x_pos/2) / f32(dimensions.x)) + half_pixel_width, input.tex_coords.y);
+
+    var uyvy = textureSample(texture, sampler_, tex_coords);
+
+    var u = uyvy.x;
+    var v = uyvy.z;
+    var y = uyvy.y;
+    if (x_pos % 2 != 0) {
+        y = uyvy.w;
+    }
```

Comment:
No idea why u and v are first. Maybe add here some links to a description of the memory layout of YUV422.

Are you sure this is a standard? There are no other possible layouts?

Reply:
Ok